packages/flutter/lib/src/painting/image_provider.dart

```diff
@@ -332,6 +332,74 @@ abstract class ImageProvider<T> {
     return ImageStream();
   }
 
+  /// Checks if this [ImageProvider] has a matching key in the
+  /// [PaintingBinding.imageCache]
+  ///
+  /// The `cache` and `configuration` parameters must not be null. If the
+  /// `handleError` parameter is null, errors will be reported to
+  /// [FlutterError.onError], and the method will return null.
+  ///
+  /// A completed return value of null indicates that an error has occurred.
+  Future<ImageCacheLocation> findCacheLocation({
+    @required ImageConfiguration configuration,
+    ImageErrorListener handleError,
```

Comment:
Why not use the Future's error reporting mechanism?

Reply:
Because the error can be thrown synchronously and won't get captured that way. This has to handle both synchronous and async errors.

Reply:
I might not be understanding what you are saying, but errors can be handled synchronously from futures:
```dart
import 'dart:async';

Future<void> doit() {
  Completer<void> completer = Completer<void>();
  completer.completeError('test');
  return completer.future;
}

void main() {
  Future<void> result = doit();
  result.catchError((err) => print(err));
}
```
prints: 'test'

Reply:
We use a special class that can behave like a synchronous call instead of a future that doesn't behave that way